internal/uvm/network.go

```diff
@@ -72,36 +72,89 @@ func (uvm *UtilityVM) removeNamespaceNICs(ns *namespaceInfo) error {
 }
 
 func (uvm *UtilityVM) addNIC(id guid.GUID, endpoint *hns.HNSEndpoint) error {
+
+	// First a pre-add. This is a guest-only request and is only done on Windows.
+	if uvm.operatingSystem == "windows" {
+
+		preAddRequest := hcsschema.ModifySettingRequest{
+			GuestRequest: guestrequest.GuestRequest{
+				ResourceType: guestrequest.ResourceTypeNetwork,
+				RequestType:  requesttype.Add,
+				Settings: guestrequest.NetworkModifyRequest{
+					AdapterInstanceId: id.String(),
+					RequestType:       requesttype.PreAdd,
+					Settings:          endpoint,
+				},
+			},
+		}
+		if err := uvm.Modify(&preAddRequest); err != nil {
+			return err
+		}
+	}
+
+	// Then the Add itself
 	request := hcsschema.ModifySettingRequest{
+		ResourcePath: path.Join("VirtualMachine/Devices/NetworkAdapters", id.String()),
 		ResourceType: resourcetype.Network,
 		RequestType:  requesttype.Add,
 		Settings: hcsschema.NetworkAdapter{
 			EndpointId: endpoint.Id,
 			MacAddress: endpoint.MacAddress,
 		},
-		GuestRequest: guestrequest.GuestRequest{
+	}
+
+	if uvm.operatingSystem == "windows" {
+		request.GuestRequest = guestrequest.GuestRequest{
+			ResourceType: guestrequest.ResourceTypeNetwork,
+			RequestType:  requesttype.Add,
+			Settings: guestrequest.NetworkModifyRequest{
+				AdapterInstanceId: id.String(),
+				RequestType:       requesttype.Add,
+			},
+		}
+	} else {
+		request.GuestRequest = guestrequest.GuestRequest{
 			ResourceType: guestrequest.ResourceTypeNetwork,
```

Comment:
This isnt supported in v1 or v2. Is this a new feature here? We have no code that can accept a modify for a nic. I guess this is a gap currently

Reply:
I am assuming it's a gap. I'm not sure how to test this though.

Reply:
Created [Microsoft/opengcs #242](https://github.com/Microsoft/opengcs/issues/242)